README.markdown

```diff
@@ -6592,6 +6610,83 @@ This API was first enabled in the `v0.9.9` release.
 
 [Back to TOC](#nginx-api-for-lua)
 
+ngx.semaphore.new
+-----------------
+**syntax:** *sem, err = ngx.semaphore.new(n)*
+
+**context:** *init_by_lua*, init_worker_by_lua*, set_by_lua*, rewrite_by_lua*, access_by_lua*, content_by_lua*, header_filter_by_lua*, body_filter_by_lua, log_by_lua*, ngx.timer.**
+
+Create a semaphore that has n resource.
+
+```lua
+
+ require "resty.core.semaphore"
+ local print = ngx.print
+ local sem,err = ngx.semaphore.new(0)
+ if not sem then
+     print("create semaphore failed: "..err)
+ end
+```
+
+[Back to TOC](#nginx-api-for-lua)
+
+ngx.semaphore.post
+------------------
+**syntax:** *ok, err = ngx.semaphore.post(sem)*
+
+**context:** *init_by_lua*, init_worker_by_lua*, set_by_lua*, rewrite_by_lua*, access_by_lua*, content_by_lua*, header_filter_by_lua*, body_filter_by_lua, log_by_lua*, ngx.timer.**
+
+The param sem is create by [ngx.semaphore.new](#ngxsemaphorenew).release one resource to a semaphore.if one light wethread or coroutine is waiting on this semaphore,then it will be wake up.
+
+```lua
+
+ require "resty.core.semaphore"
```

Comment:
I think the correct way of loading a Lua module is like this:

```lua
local semaphore = require "ngx.semaphore"
```

And then, we do

```
local sem = semaphore.new(0)
```

to instantiate a new object without using any global variables (but the `require` itself).